src/components/map/Map.tsx

```diff
@@ -1,64 +1,83 @@
 import { latLngBounds } from "leaflet";
-import React, { createRef, useContext, useEffect, useState } from "react";
-import { GeoJSON, Map as LeafletMap, TileLayer } from "react-leaflet";
-import Countries from "../../assets/countries-geo.json";
+import React, { useContext, useEffect, useState, useRef, RefObject } from "react";
+import { MapContainer, TileLayer } from "react-leaflet";
 import { AppContext } from "../../context";
-import { altStyle, getBuckets, getMapUrl, mapAltDataToFeatures, onAltEachFeature } from "../../utils/mapUtils";
+import { getBuckets, getMapUrl } from "../../utils/mapUtils";
 import Legend from "./Legend";
 import './Map.css';
+import VectorTileLayer from './VectorTileLayer.js';
+import CountriesTileLayer from "./CountriesTileLayer"
+import { layerStyle } from './MapStyle';
+
 
 export default function Map() {
-  const mapRef = createRef<LeafletMap>();
-  const geoJsonRef = createRef<GeoJSON>();
   const [state] = useContext(AppContext);
-  const [mapRecords, setMapRecords] = useState(Countries as any);
-  const [forceUpdate, setForceUpdate] = useState(Math.random());
-  const buckets = getBuckets(mapRecords.features);
+  const [mapRecords, setMapRecords] = useState(state.countries as any);
+  const buckets = getBuckets(mapRecords);
 
   useEffect(() => {
-    const importGeo = Countries as any;
-    const features = importGeo.features as GeoJSON.Feature[]
-    // We will iterate through all the features in the geoJson
-    // if they are in the country dict we will attach their aggregated data to the feature for displaying
-    importGeo.features = mapAltDataToFeatures(features, state.explore.estimateGradePrevalences);
-    setMapRecords(importGeo)
+    if(state.explore.estimateGradePrevalences.length > 0){
+      const countriesData = state.countries.map( (country : any) => {
+        const countryEstimate = state.explore.estimateGradePrevalences.find(element => element.alpha3Code === country.alpha3Code);
 
-    // we need to update the key on the GEOJSON to let react know it's time to rerender
-    setForceUpdate(Math.random())
-  }, [state.explore.estimateGradePrevalences])
+        if (countryEstimate && countryEstimate.testsAdministered) {
+          const { testsAdministered, geographicalName, numberOfStudies, localEstimate, nationalEstimate, regionalEstimate, sublocalEstimate } = countryEstimate;
+          return { ...country, properties: { testsAdministered, geographicalName, numberOfStudies, localEstimate, nationalEstimate, regionalEstimate, sublocalEstimate }}
+        }
+        return { ...country, properties: { testsAdministered: null, geographicalName: null, numberOfStudies: null, localEstimate: null, nationalEstimate: null, regionalEstimate: null, sublocalEstimate: null } }
+      })
+      setMapRecords(countriesData);
+    }
+  }, [state.explore.estimateGradePrevalences, state.countries])
 
   const bounds = latLngBounds([-90, -200], [90, 180]);
```

Comment:
I know you didn't write this code but what's the purpose of having both bounds and maxBounds

Reply:
As I understand it, If the user tries to pan the map past the bounds, the map will pull the user back to the defined bounds like an elastic. the max bounds are how much the user can stretch the map out of the normal bounds